### PR TITLE
Add data-source for Apono connector

### PR DESCRIPTION
**Pull request by asafalima**

### CODE REVIEW

File: docs/data-sources/connector.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+---
+
+# apono_connector (Data Source)
+
+Use this data source to get the ID of an Apono connector for when creating integrations.
+
+## Example Usage
+
+```terraform
+data "apono_connector" "prod_us" {
```
Comment (ofir76) [inline docs/data-sources/connector.md:16]: How it's connected to our module?
we need to be able to take the ide from the tf module that we give the account for install the connector

File: internal/provider/data_source_connector.go [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+	Name types.String `tfsdk:"name"`
+}
+
+func (d *connectorDataSource) Metadata(_ context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_connector"
+}
+
+func (d *connectorDataSource) Schema(_ context.Context, _ datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: "Use this data source to get the ID of an Apono connector for when creating integrations.",
```
Comment (ofir76) [inline internal/provider/data_source_connector.go:37]: Use this data source to get the Apono connector metadata.

File: internal/provider/data_source_connector.go [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+	var data connectorDataSourceModel
+
+	// Read Terraform configuration data into the model
+	resp.Diagnostics.Append(req.Config.Get(ctx, &data)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	tflog.Debug(ctx, "Looking for connector")
```
Comment (ofir76) [inline internal/provider/data_source_connector.go:66]: Read Apono connector

**Review by ofir76 — APPROVED** [review]
fun fun fun

**Review by ofir76 — APPROVED** [review]
fun fun fun

File: internal/provider/data_source_connector.go [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+
+// connectorDataSource defines the data source implementation for Apono connector.
+type connectorDataSource struct {
+	provider *AponoProvider
+}
+
+// connectorDataSourceModel describes the data source data model.
+type connectorDataSourceModel struct {
+	Id   types.String `tfsdk:"id"`
+	Name types.String `tfsdk:"name"`
```
[the remark this answers is not in the document]
Comment (asafalima) [reply internal/provider/data_source_connector.go:28]: And the real connector name.
Once the API include this data we can make this resource more useful by querying for connector by these fields

**Pull request merged**